docs/aragonOS.md

```diff
@@ -0,0 +1,842 @@
+# aragonOS 3.0 alpha, developer documentation
+
+*Documentation for [aragonOS](https://github.com/aragon/aragonOS) 3.0 reference implementation.
+Updated March 26th, 2018. (aragonOS v3.1.2 release)*
+
+This document provides a technical overview about the architecture and can be used
+as a specification and developer guide. For a less technically-oriented introduction
+to aragonOS 3.0, you can check the [alpha release blog post](https://blog.aragon.one/introducing-aragonos-3-0-alpha-the-new-operating-system-for-protocols-and-dapps-348f7ac92cff).
+
+## 0. Motivation
+
+aragonOS was born in our path to developing modular and upgradeable smart contracts
+to power decentralized organizations.
+
+As the project was maturing and we started abstracting more and more, we ended up
+with a pretty generic framework that can be used by any protocol or decentralized
+application that needs upgradeability.
+
+**Upgradeability** is really important when developing high-stakes systems in
+platforms that are still evolving fast. Done well, it can allow for bug fixes
+and improvements with very little disruption and not causing problems at the
+protocol level. As a history lesson, if *The DAO* had had an effective
```

Comment:
Change "if *The DAO* had had an effective" to "if *The DAO* would have had an effective"